grow/pods/documents.py

```diff
@@ -264,6 +264,13 @@ def locales(self):
             if codes is None:
                 return []
             return locales.Locale.parse_codes(codes)
+        # If `$localization` is missing, use locales inferred from document
```

Comment:
@stucox Can you take a look at this? This has the effect of *appending* locales specified by a document part's `$locale` to locales in `_blueprint.yaml`. This seems to be the desired effect based on the tests. I just don't know if it's the "correct" thing to do.

This would allow you to add a locale to a document simply by adding a document part with `$locale` versus explicitly specifying it in the base part.